tests/avatar/test_service.py

```diff
@@ -186,22 +196,61 @@ def test_png_id_is_normalized(self):
         )
 
     def test_error_checking(self):
```

Comment:
I prefer to have those tests (a) in individual test case methods and (b) using assertRaisesRegex to at least roughly check you are catching the error you are expecting to catch.

* (a) brings the advantage of overview if a single condition is violated and sense of graveness if the whole method is borked
* (b) helps with testing that it’s not a "raise RuntimeError()" at the beginning but actual different code paths being taken